README.md

```diff
@@ -7,15 +7,198 @@
+
+3. Reusing the component is possible
+
+While we focused implementing our own set of stores, some people prefer to use
+Redux, Rx, mobx or some other library for managing state and implementing
+unidirectional data flow.
+
+By storing state in an external store, we implicitly couple the component to
+whatever architecture we chose for our main application. If we're implementing
+an accordion component using Flux, it means everyone using our component will
+have to use Flux in order to interact with it (even though it might be hidden
+though the public API of the component).
+
+Hence using component state (and props) instead of stores is the preferred way
+when creating reusable components.
+
+## Connecting components using lifecycle methods
+
+So far we oftentimes "connected" components to stores by adding an event
+listener on the store in the component's lifecycle methods:
+
+```js
+class MyCounter extends React.Component {
+	componentDidMount() {
+		this.removeListener = store.addListener(count => {
+			this.setState({ count });
+		});
+	}
+	componentWillUnmount() {
+		this.removeListener();
+	}
+}
+```
+
+In `componentDidMount` as add an event listener that updates the component's
+state based on the stores state, thus triggering a re-rendering.
+
+In `componentWillUnmount` we remove the event listener. Typically `addListener`
+returns a function that — when invoked — removes the event listener from the
+store. In other cases people might subclass the `EventEmitter` class directly
+in order to implement a store, hence there are cases in which we actually have
+to call `removeListener` on the store itself.
+
+`componentDidMount` and `componentWillUnmount` are so-called lifecycle methods,
+meaning they are being run as part of the component being mounted (or
+unmounted).
+
+There are some other lifecycle methods, e.g. `componentWillMount` and
+`componentWillUpdate`. One thing to remember is that there is **no**
+`componentDidUnmount` method. This is because once a component is unmounted in
+"no longer exists" in a sense, hence doing any form of cleanup logic in there
+would be "too late".
+
+Whether or not we use `componentDidMount` instead of `componentWillMount`
+doesn't make a difference here. We can call `setState` in both.
+
+## Presentational vs Container Components
+
+When using a Flux or Flux-like architecture, it's generally speaking considered
+best-practice to avoid the number of container components.
+
+**Container components** are components that are directly **connected** to our
+store, e.g. using the `addListener` method. They are primarily concerned with
+holding state, managing it and passing it to other components using props.
+
+Container components have handler functions that dispatch actions or mutate
+state. They contain the "business logic" of our application.
+
+In single page apps, a good rule of thumb is to make each page of your
+application (or component attached to a sub-route) a container component.
```

Comment:
I'd add a small blurb about it being okay to have container components deeper in the page structure as well, as long as there aren't too many, and only when it makes sense to have them